package/lib/src/controls/date_picker.dart

```diff
@@ -0,0 +1,171 @@
+  DateTime? _value;
+  String? _state;
+
+  DateTime? get value {
+    return widget.control.attrDateTime("value");
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _value = value;
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    debugPrint("DatePicker build: ${widget.control.id}");
+
+    return StoreConnector<AppState, Function>(
+        distinct: true,
+        converter: (store) => store.dispatch,
+        builder: (context, dispatch) {
+          debugPrint("DatePicker StoreConnector build: ${widget.control.id}");
+
+          String state = widget.control.attrString("state") ?? "initState";
+          DateTime? firstDate = widget.control.attrDateTime("firstDate");
+          DateTime? lastDate = widget.control.attrDateTime("firstDate");
+          bool onChange = widget.control.attrBool("onChange", false)!;
+          String? localeString = widget.control.attrString("locale");
+          String? helpText = widget.control.attrString("helpText");
+          String? cancelText = widget.control.attrString("cancelText");
+          String? confirmText = widget.control.attrString("confirmText");
+          TextInputType keyboardType = parseTextInputType(
+              widget.control.attrString("keyboardType", "")!);
+          DatePickerMode datePickerMode = parseDatePickerMode(
+              widget.control.attrString("datePickerMode", "")!);
+          DatePickerEntryMode datePickerEntryMode = parseDatePickerEntryMode(
+              widget.control.attrString("datePickerEntryMode", "")!);
+          String? hintText = widget.control.attrString("hintText");
+
+          Locale locale;
+          if (localeString == null) {
+            locale = Localizations.localeOf(context);
+          } else {
+            locale = Locale(localeString);
+          }
+
+          void onChanged(DateTime? dateValue) {
+            debugPrint("New date: $dateValue");
+            var newState = "initState";
+            setState(() {
+              _value = dateValue;
+              _state = newState;
+            });
+            String stringValue = dateValue?.toIso8601String() ?? "null";
+            List<Map<String, String>> props = [
+              {"i": widget.control.id, "value": stringValue, "state": newState}
+            ];
+            dispatch(UpdateControlPropsAction(
+                UpdateControlPropsPayload(props: props)));
+            FletAppServices.of(context).server.updateControlProps(props: props);
+            if (onChange) {
+              FletAppServices.of(context).server.sendPageEvent(
+                  eventTarget: widget.control.id,
+                  eventName: "change",
+                  eventData: stringValue);
+            }
+            FletAppServices.of(context).server.sendPageEvent(
+                eventTarget: widget.control.id,
+                eventName: "submit",
+                eventData: "");
+          }
```

Comment:
What's the difference between `change` and `submit` events?

Reply:
change is on any text field change, submit is on select choice

Reply:
yeah, but do both events are triggered all the time when `onChange` is true?

Reply:
In [the docs](https://api.flutter.dev/flutter/material/showDatePicker.html) I see you can only grab selected date value in the future returned by `showDatePicker()`. Looks like we can implement `submit` only, right?

Reply:
yes, it was an artefact from old version

Reply:
now is on_change only available and working

Reply:
have you pushed the latest? I still see "submit" in both dart and python parts.